src/platform/audio/alaudiomanager.cpp

```diff
@@ -0,0 +1,1587 @@
+            Release_Playing_Audio(*it);
+            it = m_streamList.erase(it);
+        } else {
+            ++it;
+        }
+    }
+}
+
+/**
+ * Removes all audio with a volume of 0.
+ */
+void ALAudioManager::Remove_All_Disabled_Audio()
+{
+    for (auto it = m_globalAudioList.begin(); it != m_globalAudioList.end();) {
+        if (*it != nullptr && (*it)->openal.audio_event->Get_Volume() == 0.0f) {
+            Release_Playing_Audio(*it);
+            it = m_globalAudioList.erase(it);
+        } else {
+            ++it;
+        }
+    }
+
+    for (auto it = m_positionalAudioList.begin(); it != m_positionalAudioList.end();) {
+        if (*it != nullptr && (*it)->openal.audio_event->Get_Volume() == 0.0f) {
+            Release_Playing_Audio(*it);
+            it = m_positionalAudioList.erase(it);
+        } else {
+            ++it;
+        }
+    }
+
+    for (auto it = m_streamList.begin(); it != m_streamList.end();) {
+        if (*it != nullptr && (*it)->openal.audio_event->Get_Volume() == 0.0f) {
+            Release_Playing_Audio(*it);
+            it = m_streamList.erase(it);
+        } else {
+            ++it;
+        }
+    }
+}
+
+/**
+ * Checks if any streams have properties suggesting they are position sensitive.
+ */
+bool ALAudioManager::Has_3D_Sensitive_Streams_Playing()
+{
+    for (auto it = m_streamList.begin(); it != m_streamList.end(); ++it) {
+        if (*it != nullptr
+            && ((*it)->openal.audio_event->Get_Event_Info()->Get_Event_Type() != EVENT_MUSIC
+                || !(*it)->openal.audio_event->Get_Event_Name().Starts_With("Game_"))) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+/**
+ * Gets an audio handle for use by the video player.
+ */
+void *ALAudioManager::Get_Bink_Handle()
+{
+    // If we don't already have a playing audio for bink, make one.
+    if (m_binkPlayingAudio == nullptr) {
+        PlayingAudio *pap = new PlayingAudio;
+        Init_Playing_Audio(pap);
+        pap->openal.stopped = false;
+        pap->openal.audio_event = new AudioEventRTS("BinkHandle");
+        Get_Info_For_Audio_Event(pap->openal.audio_event);
+        pap->openal.playing_type = PAT_2DSAMPLE;
+
+        if (pap->openal.source == 0) {
+            Release_Playing_Audio(pap);
+            return nullptr;
+        }
+
+        m_binkPlayingAudio = pap;
+    }
+
+    return reinterpret_cast<void *>(m_binkPlayingAudio->openal.source);
+}
+
+/**
+ * Releases the audio handle for use by the video player.
+ */
+void ALAudioManager::Release_Bink_Handle()
+{
+    if (m_binkPlayingAudio != nullptr) {
+        Release_Playing_Audio(m_binkPlayingAudio);
+        m_binkPlayingAudio = nullptr;
+    }
+}
+
+/**
+ * Forces an audio event to be played.
+ */
+void ALAudioManager::friend_Force_Play_Audio_Event(AudioEventRTS *event)
+{
+    captainslog_warn("ALAudioManager just forwards Force_Play_Audio_Event to regular Play_Audio_Event");
```

Comment:
What is this warning good for?

Reply:
A reminder that there is no "force_Play" in ALAudioManager yet